packages/calypso-e2e/src/lib/flows/site-assembler-flow.ts

```diff
@@ -26,35 +22,20 @@ export class SiteAssemblerFlow {
 	 * @param {string} text User-visible text on the button.
 	 */
 	async clickButton( text: string ): Promise< void > {
-		await this.page.click( selectors.button( text ) );
-	}
-
-	/**
-	 * Select Header
-	 */
-	async selectHeader(): Promise< void > {
-		await this.page.getByText( 'Header' ).click();
-		const header = this.page.locator( selectors.blockRenderer ).nth( 0 );
-		await header.click();
+		await this.page.getByRole( 'button', { name: text } ).click();
 	}
 
 	/**
-	 * Select Footer
+	 * Given two parameters, type and index, selects a layout component matching
+	 * the specifications.
+	 *
+	 * @param {LayoutType} type Type of the layout component.
+	 * @param {number} index Index of the item to choose. Defaults to 0.
 	 */
-	async selectFooter(): Promise< void > {
-		await this.page.getByText( 'Footer' ).click();
-		const footer = this.page.locator( selectors.blockRenderer ).nth( 0 );
-		await footer.click();
-	}
+	async selectLayoutComponent( type: LayoutType, index = 0 ): Promise< void > {
+		await this.page.getByRole( 'button', { name: type } ).click();
+		await this.page.waitForLoadState( 'networkidle' );
```

Comment:
While we are moving away from using the `networkidle` as indicators of loaded-ness, the flip side of the coin is that the site assembler flow fires off multiple requests to the same overall endpoint differing only in the parameters that are sent:

```
endpoint&param
endpoint&param&anotherparam
endpoint&yetanotherparam
endpoint&finalvariation
```

Normally, we'd wait on a specific response from an endpoint, but this behavior makes waiting on the completion of requests from the endpoint unpredictable, which is why I've resorted to using `networkidle`.

Reply:
Thank you for addressing it!

FYI @Automattic/lego 
We might be able to reduce the requests somehow?

Reply:
When choosing headers or footers, we should not fetch rendered patterns. That's a bug that is fixed in https://github.com/Automattic/wp-calypso/pull/75861.